realm/realm-library/src/main/cpp/io_realm_internal_OsObjectSchemaInfo.cpp

```diff
@@ -37,18 +43,14 @@ JNIEXPORT jlong JNICALL Java_io_realm_OsRealmObjectSchema_nativeCreateRealmObjec
     return 0;
 }
 
-JNIEXPORT void JNICALL Java_io_realm_OsRealmObjectSchema_nativeClose(JNIEnv* env, jclass, jlong native_ptr)
+JNIEXPORT jlong JNICALL Java_io_realm_internal_OsObjectSchemaInfo_nativeGetFinalizerPtr(JNIEnv*, jclass)
 {
-    TR_ENTER_PTR(native_ptr)
-    try {
-        ObjectSchema* object_schema = reinterpret_cast<ObjectSchema*>(native_ptr);
-        delete object_schema;
-    }
-    CATCH_STD()
+    TR_ENTER()
+    return reinterpret_cast<jlong>(&finalize_object_schema);
```

Comment:
You need to keep track of the finalizer (or what ever you wish to call it). The function pointer is stored in `NativeObject` instance.